### PR TITLE
fix issue with conversion between int and void* on 64-bit

### PR DESCRIPTION
**Pull request by npitre**

### CODE REVIEW

**Review by andyross — APPROVED** [review]

File: tests/lib/ringbuffer/src/main.c [hunk 1]
```diff
@@ -129,7 +129,7 @@ static struct {
 /*entry of contexts*/
 static void tringbuf_put(void *p)
 {
-	int index = (int)p;
+	int index = (intptr_t)p;
```
Comment (nordic-krch) [inline tests/lib/ringbuffer/src/main.c]: shouldn't that be `POINTER_TO_INT`?

**Comment by nordic-krch** [comment]
Do we have any test (platform) that will find issues like that and prevent users from adding conversions which are not 64bit aware?

**Comment by npitre** [comment]
On Sun, 9 Jun 2019, Krzysztof Chruscinski wrote:

> Do we have any test (platform) that will find issues like that and prevent users from adding conversions which are not 64bit aware?

Yes. I have patches for that in the pipeline. I prefer for the fixes to 
be merged first before submitting patches that would yield lots of 
compile errors without them.

**Comment by npitre** [comment]
On Sun, 9 Jun 2019, Krzysztof Chruscinski wrote:

> -	int index = (int)p;
> +	int index = (intptr_t)p;
> 
> shouldn't that be `POINTER_TO_INT`?

Indeed.

**Review by tyrantlizard77 — APPROVED** [review]

**Review by jakub-uC — APPROVED** [review]

**Comment by andrewboie** [comment]
CI failure is spurious timing issue, we should merge this.

**Pull request merged**